rust-runtime/aws-smithy-http-server-python/src/middleware/pytests/layer.rs

```diff
@@ -0,0 +1,409 @@
+/*
```

Comment:
It feels like there is a lot of duplication in these tests, which results in long tests and a somewhat obscure purpose (i.e. what is each of them testing? It gets a bit lost).
If you extract the common bits into helper functions (e.g. the `mock::spawn_with` bits), the tests should become a bit terser and easier to digest.